typescript/packages/group-tree-plot/src/storybook/GroupTreePlot.stories.tsx

```diff
@@ -0,0 +1,60 @@
+import React from "react";
+import { GroupTreePlot } from "../GroupTreePlot";
+
+import type { EdgeMetadata, NodeMetadata } from "../types";
+
+import { exampleDatedTrees } from "../../example-data/dated-trees";
+
+/**
+ * Storybook test for the group tree plot component
+ */
+export default {
+    component: GroupTreePlot,
+    title: "GroupTreePlot/Demo",
+    argTypes: {
+        data: {
+            control: {
+                type: "object",
+            },
+        },
+    },
+};
+
+const edgeMetadataList: EdgeMetadata[] = [
+    { key: "waterrate", label: "Water Rate", unit: "m3/s" },
+    { key: "oilrate", label: "Oil Rate", unit: "m3/s" },
+    { key: "gasrate", label: "Gas Rate", unit: "m3/s" },
+    { key: "waterinjrate", label: "Water Injection Rate", unit: "m3/s" },
+    { key: "gasinjrate", label: "Gas Injection Rate", unit: "m3/s" },
+];
+
+const nodeMetadataList: NodeMetadata[] = [
+    { key: "pressure", label: "Pressure", unit: "Bar" },
+    { key: "bhp", label: "Bottom Hole Pressure", unit: "N/m2" },
+    { key: "wmctl", label: "Missing label", unit: "Unknown unit" },
+];
+
+const Template = (args) => {
+    return (
+        <GroupTreePlot
+            id={args.id}
+            datedTrees={args.datedTrees}
+            edgeMetadataList={args.edgeMetadataList}
+            nodeMetadataList={args.nodeMetadataList}
+            selectedDateTime={args.selectedDateTime}
+            selectedEdgeKey={args.selectedEdgeKey}
+            selectedNodeKey={args.selectedNodeKey}
+        />
+    );
+};
+
+export const Default = Template.bind({});
+Default.args = {
+    id: "grouptreeplot",
+    datedTrees: exampleDatedTrees,
+    edgeMetadataList: edgeMetadataList,
+    nodeMetadataList: nodeMetadataList,
+    selectedDateTime: "2018-02-01",
+    selectedEdgeKey: "oilrate",
+    selectedNodeKey: "pressure",
+};
```

Comment:
A good improvement would be to use [Storybook argTypes](https://storybook.js.org/docs/api/arg-types) for the date, edge and nodes. Eg. the date could be represented by a slider or datepicker, and the edges and nodes by an enum (https://storybook.js.org/docs/api/arg-types#controltype). This would make it easier to test this component.

Reply:
A fair point, but I all these values depend on structure of `datedTrees`, `edgeMetadataList` and `nodeMetadataList`. If these are changed, the validity of `selectedDateTime`, `selectedEdgeKey` and `selectedNodeKey` can change (i.e. the options for valid selections).

Thereby I'll stick to the basic type of string, with a descriptive comment.

As these are props of type string, I'm afraid the user will not see the connection of selected dates/keys and the actual set of possible keys/dates in the datedTrees and metadata lists.

Unless I misunderstand? If so it would be nice with an example of implementation you think of, to ensure we are on the same page.

Reply:
I'm not sure if we're talking about the same thing. Not proposing to change the component if that's what you mean.

Just to make the story easier to test you could enhance the controls in the story itself. As an example, you could add something like this:

```
Default.argTypes = {
  selectedDateTime: {
    control: "date"
  }
  selectedEdgeKey: {
    control: "select", options: getKeys(edgeMetadataList)
  }
}
```

Reply:
Probably not 100% aligned initially. I did not think you proposed to change the component, but I  was a bit unsure whether the user will loose sight of how the props actually depend on each other, if adding select where we have a list with preset options, which might become `invalid` when the user interacts with the data.

I've tested it and it looks very good, but it seems like the options for `selectedEdgeKey` and `selectedNodeKey` does not update if the `edgeMetdataList` and `nodeMetadataList` keys are changed. Is it possible to make these be updated after the user changes the keys in the lists? This is one of the examples where the user might change in the `...MetadataList`- or `datedTrees`-prop and have invalid options in the selects.

Reply:
Ok, got your point. If you still want to have a try there are some solutions using the `useArgs` hook here:

https://stackoverflow.com/questions/63708208/how-to-dynamically-mutate-args-in-storybook-v6-from-the-components-action

https://medium.com/urbandataanalytics/changing-args-values-state-in-storybook-without-usestate-hook-15cc096716f7